pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -470,12 +474,26 @@ func (c *Controller) syncCheckStates(ctx context.Context, wl *kueue.Workload, ch
 				}
 			case isCapacityRevoked(pr):
 				if workload.IsActive(wl) && !workload.IsFinished(wl) {
-					// We mark the admission check as rejected to trigger workload eviction.
+					// We mark the admission check as rejected to trigger workload deactivation.
 					// This is needed to prevent replacement pods being stuck in the pending phase indefinitely
 					// as the nodes are already deleted by Cluster Autoscaler.
 					updated = updateCheckState(&checkState, kueue.CheckStateRejected) || updated
 					updated = updateCheckMessage(&checkState, apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.CapacityRevoked).Message) || updated
 				}
+			case isBookingExpired(pr):
```

Comment:
why is the logic mostly repeated in two places? Perhaps we need some refactoring?

Reply:
Do you mean similarities between Failed and BookingExpired cases?

Reply:
more so why is the logic split in two functions `syncOwnedProvisioningRequest` and `syncCheckStates`. But it seems to be beyond the scope of this PR.

The similarity between Failed and BookingExpired implementation is also not ideal to repeat, but it looks readable right now.